freqtrade/exchange/exchange.py

```diff
@@ -622,20 +622,11 @@ def get_min_pair_stake_amount(self, pair: str, price: float, stoploss: float,
         # The value returned should satisfy both limits: for amount (base currency) and
         # for cost (quote, stake currency), so max() is used here.
         # See also #2575 at github.
-        return self._get_stake_amount_considering_leverage(
-            max(min_stake_amounts) * amount_reserve_percent,
-            leverage or 1.0
+        return self._contract_size_to_amount(
+            pair,
+            max(min_stake_amounts) * amount_reserve_percent
```

Comment:
When I look at the field amount under `fetchMarkets`, all I see is the `base` amount so I don't know why we get the max of `min_stake_amounts`

If we're using the `quote` amount, then we would need to divide by the leverage

Reply:
we're checking for both cost and amount, therefore the "real" minimum is the maximum.

for example: 
minimum amount for XRP/USDT is 10 XRP
but minimum cost for XRP is 10 USDT

Now depending on the price for 10 XRP - the higher of the 2 defines the real limit (the amount will be added as "10 XRP * current price").
This means - if current price for XRP is 2 USDT - then the minimum we're allowed to trade is for 20 USDT, therefore we need to use the maximum of the two.

It'll depend heavily on the pair - a recent example for this can be found here:
https://github.com/freqtrade/freqtrade/issues/5856#issuecomment-962621467

Reply:
I think these might be all the changes that are needed